js/frontend.js

```diff
@@ -1,41 +1,154 @@
+const formWordGoals = document.querySelector("#wordGoals");
+const formDeadline = document.querySelector("#deadline");
+const booksDiv = document.querySelector("#books");
+const bookSubmissions = document.querySelector("#book-submissions");
 
+const serverUrl = "http://localhost:3000/addbook";
 
-const form = document.querySelector('form')
-const serverUrl = 'http://localhost:3000/addbook'
+form.addEventListener("submit", e => {
+  e.preventDefault();
 
+  const formData = new FormData(form);
+  const bookTitle = formData.get("bookTitle");
+  const wordGoals = formData.get("wordGoals");
+  const deadline = formData.get("deadline");
 
-form.addEventListener('submit', (e) => {
-    e.preventDefault()
-    const formData = new FormData(form)    
-    const bookTitle = formData.get('bookTitle')
-    const wordGoals = formData.get('wordGoals')
-    const deadline = formData.get('deadline')
-    
-    const bookEntry = {
-        bookTitle,
-        wordGoals,
-        deadline
+  if (formBookTitle.value === "" || formWordGoals.value === "") {
+    alert("Fields are required.");
+  }
+
+  const bookEntry = {
+    bookTitle,
+    wordGoals,
+    deadline: new Date(deadline)
+  };
+
+  async function postData(url, data = {}) {
+    const response = await fetch(url, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json"
+      },
+      body: JSON.stringify(data)
+    });
+    return response.json();
+  }
+
+  postData(serverUrl, bookEntry)
+    .then(data => console.log(data))
+    .catch(error => console.error(error));
+
+  console.log(bookEntry);
+  formBookTitle.value = "";
+  formWordGoals.value = "";
+  document.querySelector("#books").innerHTML = "";
+  setTimeout(() => {
+    getBooks(serverUrl);
+  }, 0300);
+});
+
+async function getBooks(url) {
+  try {
+    const response = await fetch(url, {
+      method: "GET",
+      headers: {
+        "Content-Type": "application/json"
+      }
+    });
+    const data = await response.json();
+    const bookEntries = Array.from(data);
+    if (bookEntries.length > 0) {
+      createElements(bookEntries);
+      bookSubmissions.style.visibility = "visible";
     }
- 
-    async function postData(url, data = {}){
-        const response = await fetch(url, {
-            method: 'POST', 
-            headers: {
-                'Content-Type': 'application/json'
-            },
-            body: JSON.stringify(data)
-        })
-        return response.json()
+  } catch (error) {
+    console.log(error);
+  }
+}
+
+function daysDifference(date) {
+  const oneDay = 24 * 60 * 60 * 1000; // numerical value for one day to be used to count difference for deadlines
+  const dueDate = new Date(date);
+  const currentDate = new Date(Date.now());
+  const daysLeft = Math.round(Math.abs((dueDate - currentDate) / oneDay));
+  return daysLeft;
+}
+
+function createElements(data) {
+  data.forEach(book => {
+    const card = document.createElement("div");
```

Comment:
This is a lot of DOM creation code. You could clean this up by using a client side template library like handlebars: https://handlebarsjs.com/guide/#installation

Not necessary, but might be nice to get used to doing this kind of templating on the client.